### PR TITLE
3 gene tests

### PR DESCRIPTION
**Pull request by Surfboy678**
added tests:
- Array Size Gene With Mock
- Value Under The Index Gene With Mock

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -11,6 +11,7 @@
     <properties>
         <maven.compiler.source>15</maven.compiler.source>
         <maven.compiler.target>15</maven.compiler.target>
+        <java.version>11</java.version>
```
Comment (PawelDabrowski83) [inline pom.xml:14]: why do we change java version? if it is needed, should we change also in maven.compiler?

File: src/main/java/RandomProvider.java [hunk 2]
```diff
@@ -0,0 +1,3 @@
+public interface RandomProvider {
+    int getRandom();
```
Comment (PawelDabrowski83) [inline src/main/java/RandomProvider.java:2]: int getRandom(int rangeTo);
int getRandom(int rangeFrom, int rangeTo);

We want to set a range for random numbers from 0 to bound (exclusive).

File: src/main/java/main.java [hunk 3]
```diff
@@ -0,0 +1,8 @@
+
+
+public class main {
```
Comment (PawelDabrowski83) [inline src/main/java/main.java:3]: Main.class - it is convention to start class names with big letters

File: src/test/java/RandomProviderTest.java [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+@RunWith(MockitoJUnitRunner.class)
+public class RandomProviderTest {
+
+  @Test
+  public void testArraySizeGeneWithMock() {
+    // Given
+    int boundedRandomValue = ThreadLocalRandom.current().nextInt(0, 5);
```
Comment (PawelDabrowski83) [inline src/test/java/RandomProviderTest.java:17]: this is part of implementation, not needed here

File: src/test/java/RandomProviderTest.java [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+import org.junit.runner.RunWith;
+import org.mockito.runners.MockitoJUnitRunner;
+
+import java.util.concurrent.ThreadLocalRandom;
+
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+@RunWith(MockitoJUnitRunner.class)
+public class RandomProviderTest {
```
Comment (PawelDabrowski83) [inline src/test/java/RandomProviderTest.java:12]: This is RandomProviderTest. This class should not test Genes.

**Review by PawelDabrowski83 — CHANGES_REQUESTED** [review]
There are some issuses to be resolved.

File: src/test/java/GeneTest.java [hunk 6]
```diff
@@ -0,0 +1,55 @@
...
+
+    // when
+    Gene actual = new Gene(randomProviderMock);
+
+    // then
+    assertEquals(numberExpected, actual.getValues().length);
+  }
+
+  @Test
+  public void testValueUnderTheIndexGeneWithMock() {
```
Comment (PawelDabrowski83) [inline src/test/java/GeneTest.java:32]: I am missing the point of this test. We should check if every char in Gene.values is equal to expectedNumber. Why not to check it with for each loop? It should work for all elements in char.

**Review by PawelDabrowski83 — CHANGES_REQUESTED** [review]

File: pom.xml [hunk 7]
```diff
@@ -32,6 +33,12 @@
             <version>1.10.19</version>
             <scope>test</scope>
         </dependency>
+        <dependency>
+            <groupId>junit</groupId>
+            <artifactId>junit</artifactId>
+            <version>4.12</version>
```
[the remark this answers is not in the document]
Comment (Surfboy678) [reply pom.xml:39]: without this I can't import :
org.junit.Assert;
 org.junit.Test;
org.junit.runner.RunWith;

[the remark this answers is not in the document]
Comment (PawelDabrowski83) [reply pom.xml:39]: ok, let it stay and we will refactor later :)

**Review by PawelDabrowski83 — APPROVED** [review]
There are some things to be changed but I think it will work

**Review by KamilKamilS — APPROVED** [review]

**Pull request merged**